plugins/slick.autocolumnsize.js

```diff
@@ -0,0 +1,218 @@
+(function ($) {
+
+    $.extend(true, window, {
+        "Slick": {
+            "Plugins": {
+                "AutoColumnSize": AutoColumnSize
+            }
+        }
+    });
+
+    function AutoColumnSize(options) {
+        var _grid;
+        var _self = this;
+        var _handler = new Slick.EventHandler();
+        var _defaults = {
+            maxWidth: 800
+        };
+
+        var context;
+        keyCodes = {
+            'A': 65
+        };
+
+        function init(grid) {
+            options = $.extend(true, {}, _defaults, options);
+            _grid = grid;
+            maxWidth = options.maxWidth;
+
+            _handler
+                .subscribe(_grid.onHeaderCellRendered, handleHeaderCellRendered)
+                .subscribe(_grid.onBeforeHeaderCellDestroy, handleBeforeHeaderCellDestroy)
+                .subscribe(_grid.onKeyDown, handleControlKeys);
+
+            context = document.createElement("canvas").getContext("2d");
+        }
+
+        function handleHeaderCellRendered(e, args) {
+            var c = args.node;
+            $(c).on("dblclick.autosize", ".slick-resizable-handle", reSizeColumn);            
+        }
+
+        function handleBeforeHeaderCellDestroy(e, args) {           
+            var c = args.node;
+            $(c).off("dblclick.autosize", ".slick-resizable-handle", reSizeColumn);
+        }
+
+        function destroy() {
+            _handler.unsubscribeAll();
+        }
+
+        function handleControlKeys(event) {
+            if (event.ctrlKey && event.shiftKey && event.keyCode === keyCodes.A) {
+                resizeAllColumns();
+            }
+        }
+
+        function resizeAllColumns() {
+            if (_grid.getContainerNode().offsetParent === null) return;
+            var allColumns = _grid.getColumns();
+            var gridOps = _grid.getOptions();
+
+            allColumns.forEach(function (columnDef, index) {
+                if (columnDef && (!columnDef.resizable || columnDef._autoCalcWidth === true)) return;
+                var el = document.getElementById(_grid.getUID() + columnDef.id);
+                var headerWidth = getElementWidth(el);
+                var autoSizeWidth = Math.max(headerWidth, getMaxColumnTextWidth(columnDef, index)) + 1;
```

Comment:
@SatanEnglish @6pac  
Changes to the 'autoSizeWidth' value occur twice unnecessarily.
I think we can improve the code.

- Before
```
var autoSizeWidth = Math.max(headerWidth, getMaxColumnTextWidth(columnDef, index)) + 1;

// Check we are smaller than the maxWidth if provided
autoSizeWidth = Math.min(maxWidth, autoSizeWidth);
```

- After
```
// Check we are smaller than the maxWidth if provided
var autoSizeWidth = Math.min(
	maxWidth, 
	Math.max(headerWidth, getMaxColumnTextWidth(columnDef, index)) + 1
);
```